tabular/src/autogluon/tabular/models/tabular_nn/torch/tabular_nn_torch.py

```diff
@@ -620,3 +617,40 @@ def _more_tags(self):
         #  self.params_trained['batch_size'] = batch_size
         #  self.params_trained['num_epochs'] = best_epoch
         return {'can_refit_full': True}
+
+    def _valid_compilers(self):
+        return [TabularNeuralNetTorchNativeCompiler,
+                TabularNeuralNetTorchOnnxCompiler]
+
+    def _default_compiler(self):
+        return TabularNeuralNetTorchNativeCompiler
+
+    def _get_input_types(self, batch_size=None):
+        input_types = []
+        for f in self._features:
+            input_types.append((f, [batch_size, 1]))
+        return input_types
+
+    def compile(self, compiler_configs=None):
+        """
+        Compile the trained model for faster inference.
+
+        This completely overrides the compile() in AbstractModel, since we won't
+        overwrite self.model in the compilation process.
+        Instead, self.processor would be converted from sklearn ColumnTransformer
+        to its alternative counterpart.
+        """
+        assert self.is_fit(), "The model must be fit before calling the compile method."
+        if compiler_configs is None:
+            compiler_configs = {}
+        compiler = compiler_configs.get("compiler", "native")
+        batch_size = compiler_configs.get("batch_size", self.max_batch_size)
+        compiler_fallback_to_native = compiler_configs.get('compiler_fallback_to_native', False)
+
+        self._compiler = self._get_compiler(compiler=compiler,
+                                            compiler_fallback_to_native=compiler_fallback_to_native)
+        if self._compiler is not None:
+            input_types = self._get_input_types(batch_size=batch_size)
+            self.processor = self._compiler.compile(model=(self.processor, self.model),
+                                                    path=self.path,
+                                                    input_types=input_types)
```

Comment:
This looks like a lot of code dupe to sneak in the batch_size argument. Can we maybe make this more elegant to avoid code dupe?

Reply:
I put the call site of the compile() function into a `_compile` function, in order to reduce code duplication here.